### PR TITLE
knossosnet: init at 1.0.0

### PR DESCRIPTION
**Pull request by cdombroski**
## Description of changes

A multi-platform launcher for Freespace 2 Open. This C# version replaces the Python/JS version of Knossos which is no longer maintained. https://github.com/KnossosNET/Knossos.NET

Since this is a .Net program, it should build and run for Windows, Mac OS and Linux. However, this derivation was only built and tested on Linux so may need additional (runtime) libraries for other systems. Additionally, on Linux this program downloads and attempts to run Freespace 2 Open AppImage files, so appropriate support must be configured for this to work.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/kn/knossosnet/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    owner = "KnossosNET";
+    repo = "Knossos.NET";
+    rev = "v${version}";
+    hash = "sha256-nMxi6OPoyXrrjPa60DM7ZSCQ/zOuwWNuDm0rb7oO3Yc=";
+  };
+  meta = with lib; {
+    homepage = "https://github.com/KnossosNET/Knossos.NET";
+    description = "A multi-platform launcher for Freespace 2 Open";
+    license = licenses.gpl3Only;
+    mainProgram = executable;
```
Comment (SuperSandro2000) [inline pkgs/by-name/kn/knossosnet/package.nix:32]:
```suggestion
    mainProgram = "Knossos.NET";
```
according to the docs, this should be set explicitly

File: pkgs/by-name/kn/knossosnet/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  executables = [ executable ];
+  patches = [ ./targetframework.patch ];
+  runtimeDeps = [ fontconfig openal xorg.libX11 xorg.libICE xorg.libSM ];
+  src = fetchFromGitHub {
+    owner = "KnossosNET";
+    repo = "Knossos.NET";
+    rev = "v${version}";
+    hash = "sha256-nMxi6OPoyXrrjPa60DM7ZSCQ/zOuwWNuDm0rb7oO3Yc=";
+  };
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/by-name/kn/knossosnet/package.nix:28]: meta.maintainers must be set

File: pkgs/by-name/kn/knossosnet/package.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  nugetDeps = ./deps.nix;
+  executables = [ executable ];
+  patches = [ ./targetframework.patch ];
+  runtimeDeps = [ fontconfig openal xorg.libX11 xorg.libICE xorg.libSM ];
+  src = fetchFromGitHub {
+    owner = "KnossosNET";
+    repo = "Knossos.NET";
+    rev = "v${version}";
+    hash = "sha256-nMxi6OPoyXrrjPa60DM7ZSCQ/zOuwWNuDm0rb7oO3Yc=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/by-name/kn/knossosnet/package.nix:27]: a more standard formatting could look like:

```suggestion
buildDotnetModule rec {
  pname = "knossosnet";
  version = "0.2.0-RC10";

  src = fetchFromGitHub {
    owner = "KnossosNET";
    repo = "Knossos.NET";
    rev = "v${version}";
    hash = "sha256-nMxi6OPoyXrrjPa60DM7ZSCQ/zOuwWNuDm0rb7oO3Yc=";
  };

  patches = [ ./targetframework.patch ];

  nugetDeps = ./deps.nix;
  executables = [ "Knossos.NET" ];

  runtimeDeps = [ fontconfig openal xorg.libX11 xorg.libICE xorg.libSM ];
```

File: pkgs/by-name/kn/knossosnet/package.nix (hunk 2 above)
Comment (cdombroski) [reply pkgs/by-name/kn/knossosnet/package.nix:29]: Must have missed setting that while trying to figure out all the other things that had to be done correctly

**Review by SuperSandro2000 — APPROVED** [review]
didn't test this myself but the code changes LGTM

**Comment by cdombroski** [comment]
good news: this project is under active development, bad news: new versions keep getting released
hopefully having a 1.0.0 will slow the next release for a bit

**Pull request merged**